packages/headless/src/controllers/facets/range-facet/numeric-facet/headless-numeric-facet.ts

```diff
@@ -59,7 +141,13 @@ export function buildNumericFacet(
     'buildNumericFacet'
   );
 
-  dispatch(registerNumericFacet(options));
+  if (!options.generateAutomaticRanges && options.currentValues === undefined) {
+    engine.logger.error(
+      'currentValues should be specified for buildNumericFacet when generateAutomaticRanges is false.'
+    );
+  }
+
+  dispatch(registerNumericFacet(options as NumericFacetRegistrationOptions));
```

Comment:
Why is this cast needed whereas before it was not?

Reply:
Because in the new interface `currentValues` is always optional, while in `NumericFacetRegistrationOptions` it's sometimes mandatory (with the discriminating union).